### ISSUE TITLE
macOS nix-build .drv bug

### ISSUE THREAD

**Issue by Profpatsch** [target/lorri#179]
On macOS:

```
"evaluating file \'/nix/store/sv8j4gkr1l69hwyb4cm5q53g99d70q8i-closure.nix\'"
"instantiated \'lorri-keep-env-hack-shell\' -> \'/nix/store/9w5r7x944pq3ffryjvjvfb8qjiqh62vl-lorri-keep-env-hack-shell.drv\'"
"\u{1b}[31;1merror:\u{1b}[0m syntax error, unexpected \',\', expecting \')\', at \u{1b}[1m/nix/store/9w5r7x944pq3ffryjvjvfb8qjiqh62vl-lorri-keep-env-hack-shell.drv\u{1b}[0m:1:15"
```

This is a bug in nix, where `nix-build` can’t distinguish between `.drv` and `.nix` (it tries to parse the drv as a nix file). We need to move to `nix-store --realise` for that to always work.

_Originally posted by @Profpatsch in https://github.com/target/lorri/pull/178#issuecomment-543685013_

**Comment by Profpatsch** [comment]
There is a similar issue open upstream: https://github.com/NixOS/nix/issues/1958

However, it talks about the `.drv` not being in the nix store, which ours is (if I’m not overlooking something?).

**Comment by Profpatsch** [comment]
https://github.com/NixOS/nix/issues/1958#issuecomment-543775070

Fixed from nix 2.1 onwards.

Now we have the problem that Travis macOS support still uses nix 2.0.

https://github.com/target/lorri/pull/180 is a work-around for our CI, but we should really update the Travis support to use a newer nix.

**Comment by Profpatsch** [comment]
I opened an issue for updating the Travis rules: https://github.com/NixOS/nix/issues/3157
